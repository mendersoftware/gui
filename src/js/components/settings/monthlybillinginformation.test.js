import React from 'react';
import renderer from 'react-test-renderer';
import MonthlyBillingInformation from './monthlybillinginformation';

let dateMock;

describe('MonthlyBillingInformation Component', () => {
  beforeEach(() => {
    const mockDate = new Date('2019-01-01T13:00:00.000Z');
    const _Date = Date;
    global.Date = jest.fn(() => mockDate);
    global.Date.parse = _Date.parse;
    global.Date.now = _Date.now;
    global.Date.toISOString = _Date.toISOString;
    dateMock = jest.spyOn(global, 'Date').mockImplementation(() => mockDate);
  });

  afterEach(() => {
    dateMock.mockRestore();
  });

  it('renders correctly', () => {
    const mockDate = new Date();
    const tree = renderer
      .create(
        <MonthlyBillingInformation
          billingInformation={{ interactions: [], timestamp: mockDate, total: 0 }}
          isVisible={true}
          creationDate={mockDate}
          timeframe={mockDate}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
