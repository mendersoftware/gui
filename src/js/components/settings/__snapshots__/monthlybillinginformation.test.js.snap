// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlyBillingInformation Component renders correctly 1`] = `
<div
  className=""
>
  <div
    className="flexbox centered space-between margin-left-small margin-right-small"
  >
    <h2>
      Monthly usage
    </h2>
    <div
      className="explanatory-text"
    >
      Updated February 1st 2019 13:00 UTC
    </div>
  </div>
  <div
    className="flexbox centered margin-left-small"
    style={
      Object {
        "justifyContent": "flex-start",
      }
    }
  >
    <h4>
      Billing Period:
    </h4>
    <div
      className="explanatory-text margin-left"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
            />
          </svg>
        </span>
      </button>
      February 1st - February 1st 2019
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    className="billing-interaction"
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <div
      className="billing-item"
    >
      <div />
      <div>
        Quantity
      </div>
      <div>
        Unit fee
      </div>
      <div>
        Subtotal
      </div>
    </div>
  </div>
  <hr
    className="MuiDivider-root"
  />
  <div
    className="flexbox column margin-right margin-top"
    style={
      Object {
        "alignItems": "flex-end",
      }
    }
  >
    <div
      className="flexbox"
    >
      <h2
        className="explanatory-text margin-right-large"
      >
        Total
      </h2>
      <h2>
        $0
      </h2>
    </div>
  </div>
</div>
`;
